Does not build with gcc

### Describe the bug

On master, `wasmer-go` builds only with CLANG, not with gcc

### Steps to reproduce

1. Clone master (currently dfca9ab0)
1. Try building with gcc:
   ```
   $ CC=gcc go build ./...
   # github.com/wasmerio/wasmer-go/wasmer
   /usr/bin/ld: $WORK/b001/_cgo_main.o:/tmp/go-build/cgo-generated-wrappers:8: undefined reference to `wasm_limits_max_default'
   collect2: error: ld returned 1 exit status
   ```

1. Build with clang:
   ```
   CC=clang go build ./...
   ```


### Expected behavior

I expect wasmer-go to build with both gcc and clang, or at least with gcc (because it's currently documented [here][https://github.com/wasmerio/wasmer-go#install).

### Actual behavior

See error above: `undefined reference to 'wasm_limits_max_default'`

### Additional context

- OS : Fedora 33
- Go version : `go version go1.15.2 linux/amd64`
- <details>
  <summary>go env</summary>

	```
	GO111MODULE=""
	GOARCH="amd64"
	GOBIN=""
	GOCACHE="/home/vscode/.cache/go-build"
	GOENV="/home/vscode/.config/go/env"
	GOEXE=""
	GOFLAGS=""
	GOHOSTARCH="amd64"
	GOHOSTOS="linux"
	GOINSECURE=""
	GOMODCACHE="/go/pkg/mod"
	GONOPROXY=""
	GONOSUMDB=""
	GOOS="linux"
	GOPATH="/go"
	GOPRIVATE=""
	GOPROXY="direct"
	GOROOT="/usr/lib/golang"
	GOSUMDB="off"
	GOTMPDIR=""
	GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
	GCCGO="gccgo"
	AR="ar"
	CC="gcc"
	CXX="g++"
	CGO_ENABLED="1"
	GOMOD="/workspaces/wasmer-go/go.mod"
	CGO_CFLAGS="-g -O2"
	CGO_CPPFLAGS=""
	CGO_CXXFLAGS="-g -O2"
	CGO_FFLAGS="-g -O2"
	CGO_LDFLAGS="-g -O2"
	PKG_CONFIG="pkg-config"
	GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build788586221=/tmp/go-build -gno-record-gcc-switches"
	```
	</details>

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -16,10 +16,8 @@ jobs:
         target:
           - id: 'linux-amd64'
             os: 'ubuntu-18.04'
-            cc: 'clang-6.0'
           - id: 'darwin-amd64'
             os: 'macos-latest'
-            cc: 'clang'
         go: [1.14]
       fail-fast: true
 
@@ -53,15 +51,11 @@ jobs:
         shell: bash
         run: |
           just test
-        env:
-          CC: ${{ matrix.target.cc }}
 
       - name: Run all the examples
         shell: bash
         run: |
           just examples
-        env:
-          CC: ${{ matrix.target.cc }}
 # Skipped for now because of Github banning the API request from actions.
 # You can find more info in this PR: https://github.com/wasmerio/wasmer-go/pull/118#issuecomment-588487544
 #       - name: Test bazel build

--- a/wasmer/limits.go
+++ b/wasmer/limits.go
@@ -1,10 +1,16 @@
 package wasmer
 
 // #include <wasmer_wasm.h>
+//
+// uint32_t limit_max_unbound() {
+//     return wasm_limits_max_default;
+// }
 import "C"
 import "runtime"
 
-const LimitMaxUnbound = uint32(C.wasm_limits_max_default)
+func LimitMaxUnbound() uint32 {
+	return uint32(C.limit_max_unbound())
+}
 
 type Limits struct {
 	_inner C.wasm_limits_t

--- a/wasmer/limits_test.go
+++ b/wasmer/limits_test.go
@@ -6,7 +6,7 @@ import (
 )
 
 func TestLimitMaxDefault(t *testing.T) {
-	assert.Equal(t, LimitMaxUnbound, uint32(0xffffffff))
+	assert.Equal(t, LimitMaxUnbound(), uint32(0xffffffff))
 }
 
 func TestLimits(t *testing.T) {

